common/src/main/java/io/netty/util/concurrent/FastThreadLocalThread.java

```diff
@@ -71,6 +77,10 @@ public FastThreadLocalThread(ThreadGroup group, Runnable target, String name, lo
      * Note that this method is for internal use only, and thus is subject to change at any time.
      */
     public final InternalThreadLocalMap threadLocalMap() {
+        if (this != Thread.currentThread() && logger.isWarnEnabled()) {
+            logger.warn(new Exception("It's not thread-safe to get 'threadLocalMap' " +
+                    "which doesn't belong to the caller thread"));
```

Comment:
```suggestion
            logger.warn("It's not thread-safe to get 'threadLocalMap' " +
                    "which doesn't belong to the caller thread", new RuntimeException());
```

If we would have min of java9 we could have used `IllegalCallerException`...